src/app/CommandHandler.h

```diff
@@ -190,9 +209,56 @@ class CommandHandler : public Messaging::ExchangeDelegate
     CHIP_ERROR AddClusterSpecificFailure(const ConcreteCommandPath & aCommandPath, ClusterStatus aClusterStatus);
 
     Protocols::InteractionModel::Status ProcessInvokeRequest(System::PacketBufferHandle && payload, bool isTimedInvoke);
-    CHIP_ERROR PrepareCommand(const ConcreteCommandPath & aCommandPath, bool aStartDataStruct = true);
+
+    /**
+     * This will add a new CommandDataIB element into InvokeResponses. It will put the
+     * aResponseCommandPath into CommandPath element within CommandDataIB.
+     *
+     * This call will fail if CommandHandler is already in the middle of building/sending
+     * InvokeResponseMessage.
+     *
+     * Upon success, the caller is expected to call `FinishCommand` once they have added
+     * all Data into Fields element of CommandDataIB.
+     *
+     * @param [in] aRequestCommandPath the concrete path of the command we are responding to.
+     * @param [in] aResponseCommandPath the concrete response path that we are sending to Requester.
+     * @param [in] aStartDataStruct starts the TLV container for the CommandFields element within
+     *             CommandDataIB.
+     */
+    CHIP_ERROR PrepareInvokeResponseCommand(const ConcreteCommandPath & aRequestCommandPath,
+                                            const ConcreteCommandPath & aResponseCommandPath, bool aStartDataStruct = true);
+    [[deprecated("PrepareCommand now needs the requested command path. Please use PrepareInvokeResponseCommand")]] CHIP_ERROR
+    PrepareCommand(const ConcreteCommandPath & aCommandPath, bool aStartDataStruct = true);
+
+    /**
+     * Finishes the CommandDataIB element within the InvokeResponses.
+     *
+     * Caller must have first successfully called `PrepareInvokeResponseCommand`.
+     *
+     * @param [in] aEndDataStruct end the TLV container for the CommandFields element within
+     *             CommandDataIB.
```

Comment:
Are there any error returns we are sure will happen? Suggest describing some of the more important errors than can arise.